data/transactions/logic/eval.go

```diff
@@ -214,6 +217,7 @@ func Eval(program []byte, params EvalParams) (pass bool, err error) {
 	cx.EvalParams = params
 	cx.stack = make([]stackValue, 0, 10)
 	cx.program = program
+	cx.programHash = crypto.HashObj(Program(program))
```

Comment:
I slightly wish this was lazy since I bet we won't need it for 90% of programs, but I'm not sure we need that optimization yet.

Reply:
Sounds good--maybe we can pass it in? (Since we'll have computed upstream at some point, either when decommitting to an escrow account or when verifying some delegated LogicSig.)